pkg/operatormetrics/collector_test.go

```diff
@@ -104,5 +108,34 @@ var _ = Describe("Collector", func() {
 			Expect(metricCounter.Desc().String()).To(ContainSubstring(testCounterOpts.Name))
 			Expect(metricCounter.Desc().String()).To(ContainSubstring("constLabels: {important_info=\"xpto\"}"))
 		})
+
+		It("should collect metrics with custom timestamps", func() {
+			counter := NewCounter(testCounterOpts)
+
+			collector := Collector{
+				Metrics: []Metric{counter},
+				CollectCallback: func() []CollectorResult {
+					return []CollectorResult{
+						{Metric: counter, Labels: nil, Value: 5, Timestamp: time.UnixMilli(1000)},
+					}
+				},
+			}
+
+			err := RegisterCollector(collector)
+			Expect(err).NotTo(HaveOccurred())
+
+			ch := make(chan prometheus.Metric, 1)
+			go collector.Collect(ch)
+
+			metricCounter := <-ch
+
+			Expect(metricCounter.Desc().String()).To(ContainSubstring(testCounterOpts.Name))
+
+			dto := &io_prometheus_client.Metric{}
+			err = metricCounter.Write(dto)
+			Expect(err).NotTo(HaveOccurred())
+
```

Comment:
I think no need in empty line here

Reply:
i wanted to separate the variable creation from the value validation